compiler/src/dotty/tools/dotc/transform/CheckUnused.scala

```diff
@@ -281,24 +282,27 @@ class CheckUnused private (phaseMode: CheckUnused.PhaseMode, suffix: String, _ke
 
   /** Do the actual reporting given the result of the anaylsis */
   private def reportUnused(res: UnusedData.UnusedResult)(using Context): Unit =
-    res.warnings.toList.sortBy(_.pos.line)(using Ordering[Int]).foreach { s =>
+    res.warnings.toList.sortBy{
+      case us: UnusedSymbol.ImportSelector => us.pos.line
+      case us: UnusedSymbol.Symbol => us.tree.sourcePos.line
+      }(using Ordering[Int]).foreach { s =>
       s match
-        case UnusedSymbol(t, _, WarnTypes.Imports) =>
-          report.warning(s"unused import", t)
-        case UnusedSymbol(t, _, WarnTypes.LocalDefs) =>
-          report.warning(s"unused local definition", t)
-        case UnusedSymbol(t, _, WarnTypes.ExplicitParams) =>
-          report.warning(s"unused explicit parameter", t)
-        case UnusedSymbol(t, _, WarnTypes.ImplicitParams) =>
-          report.warning(s"unused implicit parameter", t)
-        case UnusedSymbol(t, _, WarnTypes.PrivateMembers) =>
-          report.warning(s"unused private member", t)
-        case UnusedSymbol(t, _, WarnTypes.PatVars) =>
-          report.warning(s"unused pattern variable", t)
-        case UnusedSymbol(t, _, WarnTypes.UnsetLocals) =>
-          report.warning(s"unset local variable, consider using an immutable val instead", t)
-        case UnusedSymbol(t, _, WarnTypes.UnsetPrivates) =>
-          report.warning(s"unset private variable, consider using an immutable val instead", t)
+        case UnusedSymbol.ImportSelector(srcPos, _) =>
+          report.warning(UnusedSymbolMessage.imports(srcPos.sourcePos))
```

Comment:
It seems like method `warning` overrides position
```scala
  def warning(msg: Message)(using Context): Unit =
    warning(msg, NoSourcePosition)
```

Reply:
good catch :+1: fixed by passing the position explicite as the second argument.